queue/queue.py

```diff
@@ -3,13 +3,14 @@ def __init__(self):
     self.size = 0
     # what data structure should we
     # use to store queue elements?
-    self.storage = 
+    self.storage = []
 
   def enqueue(self, item):
-    pass
+    self.storage.append(item)
   
   def dequeue(self):
-    pass
+    if self.storage:
+      return self.storage.pop(0)
```

Comment:
For example of the above critique:

Since we are using array/list, popping an item off the front forces all other items in the structure to be re-index'd (a linear time operation).  If we were using a LinkedList(), this would be a constant time operation instead.